preprocessors/openstreetmap/main.py

```diff
@@ -97,29 +96,28 @@ def get_map_data():
             POD1 = allot_intersection(
                 processed_OSM_data,
                 intersection_record_updated)
-        POIs = enlist_POIs(POD1, amenity)
+        POIs, new_POIs = enlist_POIs(POD1, amenity, OSM_data)
         if processed_OSM_data is not None and len(processed_OSM_data) != 0:
             response = OSM_preprocessor(processed_OSM_data, POIs, amenity)
             response = {
```

Comment:
Not necessarily. This is for the new issue raised by Vennisa which we discussed on the possibility of having a new structure that would not necessarily require her to loop through all the POIs before accessing them. I've checked on her this morning, hopefully to meet today to finalize things if this is the way to go.

Reply:
> Why are we overwriting `response` in the next line? Are `POIs` and `amenity` ever used, or just `new_POIs` and `new_amenity`?

`POIs and ` amenity` are used throughout the code. The new variables are mere duplicates of them, but in slightly different formats. I need to run the new formats with Vennisa before this can be concluded.